src/nagini_translation/analyzer.py

```diff
@@ -1026,6 +1064,15 @@ def convert_type(self, mypy_type, node) -> PythonType:
         elif self.types.is_normal_type(mypy_type):
             return self._convert_normal_type(mypy_type)
         elif self.types.is_tuple_type(mypy_type):
+            # Handle case where ADT (class) is typed as tuple
+            if hasattr(mypy_type, 'fallback'):
+                try:
+                    fallback_type = self.convert_type(mypy_type.fallback, node)
+                    if isinstance(fallback_type, PythonClass) and fallback_type.is_adt:
+                        return fallback_type
+                except UnsupportedException:
+                    pass
```

Comment:
Can we really just drop exceptions here? If yes, IMO there should be a comment that explains why.

Reply:
Comments added.